src/components/rest-client/RestClientForm.tsx

```diff
@@ -0,0 +1,234 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import { useTranslations } from "next-intl";
+import {
+  BodyType,
+  CustomResponseBody,
+  ErrorResponseBody,
+  HeadersType,
+  HttpMethod,
+} from "../../types/rest-client";
+import {
+  beautify,
+  createRequestUrl,
+  parseUrl,
+} from "../../utils/rest-client-utils";
+import ErrorModal from "@/components/ui/ErrorModal.tsx";
+import Body from "@/components/rest-client/Body.tsx";
+import HeadersTable from "@/components/rest-client/HeadersTable.tsx";
+import CodeGenerator from "@/components/rest-client/CodeGenerator.tsx";
+
+export default function RestClientForm() {
+  const rest = useTranslations("RestClient");
+  const modal = useTranslations("Modal");
+
+  const [url, setUrl] = useState("");
+  const [method, setMethod] = useState<HttpMethod>("GET");
+  const [headers, setHeaders] = useState<{ key: string; value: string }[]>([]);
+  const [body, setBody] = useState<string>("");
+  const [response, setResponse] = useState<string>("");
+  const [loading, setLoading] = useState(false);
+  const [status, setStatus] = useState<string>("");
+  const [error, setError] = useState<Error | null>(null);
+  const [showCodeGen, setShowCodeGen] = useState(false);
```

Comment:
can this be use as useReducer instead?

Reply:
Also we could potentially create a custom hook useRestClient for example to reuse it